cmd/erasure-server-pool-rebalance.go

```diff
@@ -133,6 +133,11 @@ func (z *erasureServerPools) initRebalanceMeta(ctx context.Context, buckets []st
 	}, len(z.serverPools))
 	var totalCap, totalFree uint64
 	for _, disk := range si.Disks {
+		// Ignore invalid.
+		if len(diskStats) <= disk.PoolIndex {
```

Comment:
I don't think this will work:

we have 'unrecognizedDisk' which is used in place for regular xlStorage in case of an order issue. disk.PoolIndex will be -1 and len(diskStats) will never be less or equal to disk.PoolIndex.

I think what you need to do is check if disk.PoolIndex == -1  

But also add a new code here `cmd/erasure.go, line 190` to return `pool, set, disk = -1, -1, -1` (you will understand why)